### ISSUE TITLE
Deprecation: scope variable 'bar' assigned to non-scope parameter '_param_2' calling formattedWrite

### ISSUE THREAD

**Issue by dlangBugzillaToGithub** [dlang/dmd#20209]
### CC List

* Paul Backus

### Description

````markdown
```
import std.format;
void foo(scope immutable(char)[][] bar) @safe {
	static struct Result {
		void put(char c) @safe {}
	}
	Result().formattedWrite("%s", bar);
}
```
The interesting thing about this code is that the deprecation vanishes if bar is `scope immutable(ubyte)[][]` instead. Copying formattedWrite into this module seems to make the warning go away as well, so I wasn't able to reduce this example any further.
````

**Comment by dlangBugzillaToGithub** [comment]
#### snarwin+bugzilla commented on 2023-03-02T19:30:09Z



````markdown
Tested this with DMD/Phobos master and it compiles with no errors, so this should be fixed in the 2.103.0 release.
````

**Comment by thewilsonator** [comment]
Works for me
